stablehlo/dialect/Base.cpp

```diff
@@ -176,5 +176,151 @@ Attribute boundsToEncoding(Attribute prototype, ArrayRef<int64_t> bounds) {
   return dialect->createBoundedAttr(bounds);
 }
 
+// Inference rules to concat dimensions with bounds (lhs/rhs are commutative):
+//       Dim of lhs     Dim of rhs      Infer
+//  c0:  X              Y               X+Y
+//  c1:  X              ?               ?
+//  c2:  X              ?, B            ?, X+B
+//  c3:  ?              ?               ?
+//  c4:  ?              ?, B            ?
+//  c5:  ?, B1          ?, B2           ?, B1+B2
+static std::pair<int64_t, int64_t> inferConcatenatedDimAndBound(
+    int64_t leftSize, int64_t rightSize, int64_t leftBound,
+    int64_t rightBound) {
+  bool isLeftStaticDim = leftSize != ShapedType::kDynamicSize;
```

Comment:
how about something like,

size = ShapedType::kDynamicSize;
bound = ShapedType::kDynamicSize;

```
if (isLeftStaticDim && isRightStaticDim) {
  size = leftSize + rightSize;
} else {
  leftSizeOrBound = isLeftStaticDim ? leftSize : leftBound;
  rightSizeOrBound = isRightStaticDim ? rightSize : rightBound;
  if (leftSizeOrBound != ShapedType::kDynamicSize && 
      rightSizeOrBound != ShapedType::kDynamicSize)
    bound = leftSizeOrBound + rightSizeOrBound;
}
```

Reply:
Nice code! Replaced and thanks.